front/src/App.tsx

```diff
@@ -6,22 +6,47 @@ import SpellbookContainer, {
 } from "components/spellbookContainer/SpellbookContainer";
 import { Character } from "types";
 import { ClassName } from "common/character";
+import useStateWithLocalStorage from "hooks/useStateWithLocalStorage";
 
 const INITIAL_CHARACTER: Character = {
     name: "Josh Mann",
     class: ClassName.Sorcerer,
     spellCastingModifier: 0,
     showSpellSaveDC: true,
+    id: "",
 };
 
+const CHARACTERS_KEY = "characters";
+
 function App() {
-    const [character, setCharacter] = useState<Character>(INITIAL_CHARACTER);
+    const fallbackCharacter: Character = {
+        ...INITIAL_CHARACTER,
+        id: crypto.randomUUID(),
+    };
+
+    const [characters, setCharacters] = useStateWithLocalStorage<Character[]>(
+        CHARACTERS_KEY,
+        [fallbackCharacter]
+    );
+    const [currentCharacterID] = useState<string>(characters[0].id);
+
+    const currentCharacter: Character =
+        characters.find((character) => character.id == currentCharacterID) ||
+        fallbackCharacter;
```

Comment:
We're always garunteed a fallback character, as we set one for the `defaultValue`.

```suggestion
    const currentCharacter = useMemo(() => {
            return characters.find((character) => character.id == currentCharacterID)
        },
        [characters, currentCharacterId]
    );
```